### PR TITLE
Extend logging and verbosity options

### LINKED ISSUE #2627: [Feature request] Less verbose output
I propose we that allow a less verbose mode. 

The existing way of Fantomas deals with this problem is being very verbose doing `dotnet fantomas --recurse .` where is each file processed is being list..  It would be helpful to being less verbose not print any info about unchanged files.

## Pros and Cons 

The advantages of making this adjustment to Fantomas are less nosiy output allows user to focus on important things.

The disadvantages of making this adjustment to Fantomas are it is a change and someone might be dependent on current behavior of the cli tool. 
It might also might make ignored files harder to debug.

## Extra information

Estimated cost (XS, S, M, L, XL, XXL): S

Related suggestions: (put links to related suggestions here)

## Affidavit (please submit!)

Please tick this by placing a cross in the box:
* [X] I have read the [Contribution Guidelines](https://github.com/fsprojects/fantomas/blob/main/CONTRIBUTING.md).
* [X] I have [searched both open and closed suggestions on this site](https://github.com/fsprojects/fantomas/issues) and believe this is not a duplicate

Please tick all that apply:
* [ ] This is not a breaking change to Fantomas
* [X] I or my company would be willing to help implement and/or test this
* [ ] This suggestion is part of the [Microsoft style guide](https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting) (please add a link to section if so)
* [ ] This suggestion is part of the [G-Research style guide](https://github.com/G-Research/fsharp-formatting-conventions) (please add a link to section if so)

### PR DESCRIPTION
**Pull request by dawedawe**
To show the general direction and maybe get some feedback, here's an early draft for the logging feature:
`--verbosity d` should be the same as in the past with the addition of the count of processed files in the end
`--verbosity n` should be pretty quiet

Fixes #2627

### CODE REVIEW

File: src/Fantomas/IgnoreFile.fs [hunk 1]
```diff
@@ -90,5 +91,5 @@ module IgnoreFile =
             try
                 ignoreFile.IsIgnored fullPath
             with ex ->
-                printfn "%A" ex
+                sprintf "%A" ex |> stdlog
```
Comment (nojaf) [inline src/Fantomas/IgnoreFile.fs:94]: This probably should be an `elog`.

File: src/Fantomas/Program.fs [hunk 2]
```diff
@@ -35,12 +37,13 @@ type Arguments =
...
                     "Input paths: can be multiple folders or files with %s extension."
                     (Seq.map (fun s -> "*" + s) extensions |> String.concat ",")
+            | Verbosity _ -> "Set the verbosity level. Allowed values are n[ormal] and d[etailed]."
 
 let time f =
     let sw = Diagnostics.Stopwatch.StartNew()
     let res = f ()
     sw.Stop()
-    printfn "Time taken: %O s" sw.Elapsed
+    sprintf "Time taken: %O s" sw.Elapsed |> stdlog
```
Comment (nojaf) [inline src/Fantomas/Program.fs:46]: I guess all the `stdlog` calls can be interpolated strings instead.
The extra pipe seems redundant.

File: src/Fantomas/Program.fs [hunk 3]
```diff
@@ -166,39 +169,39 @@ let readFromStdin (lineLimit: int) =
         else
             Some(input)
 
-let private reportCheckResults (output: TextWriter) (checkResult: Format.CheckResult) =
+let private reportCheckResults (checkResult: Format.CheckResult) =
     checkResult.Errors
     |> List.map (fun (filename, exn) -> sprintf "error: Failed to format %s: %s" filename (exn.ToString()))
-    |> Seq.iter output.WriteLine
+    |> Seq.iter stdlog
```
Comment (nojaf) [inline src/Fantomas/Program.fs:175]: error log?

File: src/Fantomas/Program.fs [hunk 4]
```diff
@@ -359,31 +392,37 @@ let main argv =
...
+            | InputPath.Multiple(files, folders), OutputPath.NotKnown ->
+                let n = filesAndFolders force files folders
+                logGrEqDetailed verbosity (sprintf "Processed files: %d" n)
             | InputPath.Multiple _, OutputPath.IO _ ->
-                eprintfn "Multiple input files are not supported with the --out flag."
+                elog "Multiple input files are not supported with the --out flag."
                 exit 1
         with exn ->
-            printfn "%s" exn.Message
+            stdlog exn.Message
```
Comment (nojaf) [inline src/Fantomas/Program.fs:425]: error log?

**Review by nojaf — COMMENTED** [review]
This looks good so far!
I would be a little more consistent when using the logger functions. Just always go with an interpolated string. But that is really a nit.

I also think we should keep this for v6. Just to be on the safe side and get some user feedback.

**Comment by dawedawe** [comment]
> This looks good so far! I would be a little more consistent when using the logger functions. Just always go with an interpolated string. But that is really a nit.
> 
> I also think we should keep this for v6. Just to be on the safe side and get some user feedback.

Thanks for the review. Waiting for v6 is fine for me.

**Comment by nojaf** [comment]
I created a https://github.com/fsprojects/fantomas/tree/v6.0 branch which you can use to target this PR.

**Comment by dawedawe** [comment]
If this is okay, I'd like to merge this, so Jimmy and me run into less conflicts with our next steps.
I'm thinking about the changes needed to report in the end how many files failed and how many were good...

**Review by nojaf — APPROVED** [review]
I think this is indeed ok to have as is now and revisit later.
Some food for thought:
- Do we not want to show the user which file was formatted? I think this makes sense in normal logging.
- Should we free up `-v` for verbosity? It is current used for version. It might make sense to use it for verbosity instead.
-  We will need to think about the format of logging. Seeing `[15:21:11 INF]` might be a bit misleading when `detailed` is active.

**Comment by dawedawe** [comment]
> I think this is indeed ok to have as is now and revisit later. Some food for thought:
> 
>     * Do we not want to show the user which file was formatted? I think this makes sense in normal logging.
> 
>     * Should we free up `-v` for verbosity? It is current used for version. It might make sense to use it for verbosity instead.
> 
>     * We will need to think about the format of logging. Seeing `[15:21:11 INF]` might be a bit misleading when `detailed` is active.

Thanks a lot, I'll work on these.

**Pull request merged**